### ISSUE TITLE
Test failure after update to ROCm 2.8 - test_gpu_quantization

### ISSUE THREAD

**Issue by mvermeulen** [ROCmSoftwarePlatform/AMDMIGraphX#379]
After updating to ROCm 2.8, I see the following failure
```
 67/190 Test  #36: test_gpu_quantization .................................................***Failed  Error regular expression found in output. Regex=[FAILED] 17.09 sec
[   RUN    ] target_copy
void target_copy()
/home/mev/source/AMDMIGraphX/test/gpu/quantization.cpp:65:
    FAILED: migraphx::verify_range(cpu_result, gpu_result) 0
CMake Error at gdb/test_test_gpu_quantization/run.cmake:12 (message):
  Test failed
```

Built a debuggable version and see the following using gdb - values in the vectors look different.
```
(gdb) r
Starting program: /home/mev/source/AMDMIGraphX/build-2019-09-27-rocm28-g/bin/test_gpu_quantization 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffcd91e700 (LWP 34357)]
[   RUN    ] target_copy
[New Thread 0x7fffcca58700 (LWP 34360)]
void target_copy()
/home/mev/source/AMDMIGraphX/test/gpu/quantization.cpp:65:
    FAILED: migraphx::verify_range(cpu_result, gpu_result) 0

Thread 1 "test_gpu_quanti" received signal SIGABRT, Aborted.
0x00007fffde0f3428 in __GI_raise (sig=sig@entry=6)
    at ../sysdeps/unix/sysv/linux/raise.c:54
54	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  0x00007fffde0f3428 in __GI_raise (sig=sig@entry=6)
    at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007fffde0f502a in __GI_abort () at abort.c:89
#2  0x000000000040cf24 in test::failed<test::lhs_expression<bool const&>, void (*)()> (x=..., msg=0x42b6e4 "migraphx::verify_range(cpu_result, gpu_result)", 
    func=<optimized out>, file=<optimized out>, line=65, 
    f=0x7fffde0f4ec0 <__GI_abort>)
    at /home/mev/source/AMDMIGraphX/test/include/test.hpp:140
#3  0x000000000040aca5 in target_copy ()
    at /home/mev/source/AMDMIGraphX/test/gpu/quantization.cpp:65
#4  0x0000000000425ff9 in std::function<void ()>::operator()() const (
    this=0x1209be0)
    at /usr/lib/gcc/x86_64-linux-gnu/7.4.0/../../../../include/c++/7.4.0/bits/std_function.h:706
#5  test::run_test_case(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void ()> const&) (
    name="target_copy", f=...)
    at /home/mev/source/AMDMIGraphX/test/include/test.hpp:219
#6  0x000000000041093f in test::run (argc=<optimized out>, 
    argv=<optimized out>)
    at /home/mev/source/AMDMIGraphX/test/include/test.hpp:232
#7  0x0000000000410576 in main (argc=34353, argv=0x8631)
    at /home/mev/source/AMDMIGraphX/test/gpu/quantization.cpp:128
(gdb) frame 3
#3  0x000000000040aca5 in target_copy ()
    at /home/mev/source/AMDMIGraphX/test/gpu/quantization.cpp:65
65	        EXPECT(migraphx::verify_range(cpu_result, gpu_result));
(gdb) p cpu_result
$1 = std::vector of length 9, capacity 9 = {0.4375, 0.75, 0.1875, 0.25, 0.25, 
  0.9375, -0.75, 0.375, 0.1875}
(gdb) p gpu_result
$2 = std::vector of length 9, capacity 9 = {-2.78713754e+38, -2.78713754e+38, 
  -1.71802739e+38, 2.28124189, -nan(0x5eff01), 0.9375, -0.75, 0.375, 0.1875}
(gdb) 
```

**Comment by mvermeulen** [comment]
Added some debugging output statements
```
[   RUN    ] target_copy
p = 
y = @param:y -> float_type, {3, 3}, {3, 1}
x = @param:x -> float_type, {3, 3}, {3, 1}
@0 = add(x,y) -> float_type, {3, 3}, {3, 1}

m[x] = 0.4375, 0.75, 0.1875, 0.25, 0.25, 0.9375, -0.75, 0.375, 0.1875
cpu_result = 0.4375,0.75,0.1875,0.25,0.25,0.9375,-0.75,-0.75,0.1875
m[x] = 0.4375, 0.75, 0.1875, 0.25, 0.25, 0.9375, -0.75, 0.375, 0.1875
gpu_result = -2.79134e+38,-2.79134e+38,-nan,-9.22335e+18,-nan,0.9375,-0.75,-0.75,-0.185049
void target_copy()
/home/mev/source/AMDMIGraphX/test/gpu/quantization.cpp:86:
    FAILED: migraphx::verify_range(cpu_result, gpu_result) 0
Aborted (core dumped)
```

One thing that does confuse me on the the code is where is the parameter m["y"] initialized?

**Comment by scxiao** [comment]
I observed the same error and have a fix in the PR #378.
